### DISCUSSION TITLE
Prevent line break

### DISCUSSION THREAD

**Discussion by PgLoLo** [typst/typst#4372] [Q&A] [answered]
In some places I wont to prevent typst to automatically line break the text. Examples: 
1. I have an inline formula "... $1 times N times 1$ ..." and I want to forbid typst to put line breaks inside of it
2. I have some ordinary text that I do not want to line break. For example, "... TT-ranks ...": I want to forbid line break between "TT-" and "ranks".
3. In general, I may have some block of text that I want to be guaranteed in one line

Is there any way to achieve this?

**Comment by flokl** [comment, marked as answer]
Hi, you can enclose the content with `#box[...]`

**Comment by Enivex** [comment]
> Otherwise, you can insert a [ZWJ](https://en.wikipedia.org/wiki/Zero-width_joiner).
> Maybe, there are better solutions, but the two above work.
> It seems there is also the ["word joiner"](https://en.wikipedia.org/wiki/Word_joiner) character.

WJ is the correct one here, ZWJ does not do what OP is looking for.

That being said, the box is simpler unless you're looking to remove only a very specific breaking point.

**Reply by rben01** [reply]
I had to use `\u{2060}$""^,$\u{2060}` to prevent the line from breaking at the comma, which is probably not desired. `#box[]` didn't work in my case.

**Reply by joonas-si** [reply]
`#box[]` didn't work for me either... it prevented the line break but also the text enclosed in teh brackets was not in line with the rest of the text anymore... IN the image testname and the registered trademark should not be split into two line, but adding the `#box[]` around takes testname out of position. Did anybody encounter similar problems.
<img width="1438" height="135" alt="image" src="https://github.com/user-attachments/assets/07008285-61d0-436b-9807-6e45de5753ca" />
